File: Mendoza, Santiago RJ/Javascript/TSLeccion07/Persona.js
class Persona{

    static contadorPersona = 0;

    constructor(nombre, apellido, edad){
        this._idPersona = ++Persona.contadorPersona;
        this._nombre = nombre;
        this._apellido = apellido;
        this._edad = edad;
    }

    get idPersona(){
        return this.idPersona;
    }

    get nombre(){
        return this._nombre;
    }

    set nombre(nombre){
        this._nombre = nombre;
    }

    get apellido(){
        return this._apellido;
    }

    set apellido(apellido){
        this._apellido = apellido;
    }

    get edad(){
        return this._edad;
    }

    set edad(edad){
        this._edad = edad;
    }

    toString(){
        return this._idPersona+' '+this._nombre+' '+this._apellido+' '+this._edad
    }
}

File: Semana 03 - 17-04/Troncoso, Braian Axel/Java/Lección02/BloquesInicializacion/src/domain/Persona.java
package domain;
public class Persona {
    private final int idPersona;
    private static int contadorPersona;

    static { // Bloque de inicializacion estático
        System.out.println("Ejecución bloque estático");
         ++Persona.contadorPersona;
        //idPersona = 10; No es un atributo estático, por esto tenemos un error
    } // Dentro del contexto estático no podemos acceder al operador this.


    { //Bloque de inicializacion NO estático (contexto dinámico)
        System.out.println("Ejecución del bloque NO estático");
        this.idPersona = Persona.contadorPersona++; //Incrementamos el atributo
    }
    // Los bloques de inicialización se ejecutan antes del constructor(de nuestra clase)

    public Persona(){
        System.out.println("Ejecución del constructor");
    }

    public int getIdPersona(){
        return this.idPersona;
    }

    @Override
    public String toString() {
        final StringBuilder sb = new StringBuilder("Persona{");
        sb.append("idPersona=").append(idPersona);
        sb.append('}');
        return sb.toString();
    }
}
